### PR TITLE
Change the way benchmarks are launched

### LINKED ISSUE #4: Enos bench
Based on the previous version, I've prototyped a new way to launch benchs : 

https://github.com/msimonin/kolla-g5k/commit/c23d41a8a3ea04d6a12ac30d86a3ef8062348bb9

Let's see if it fit our needs  :)

### PR DESCRIPTION
**Pull request by msimonin**
Change the way benchmarks are launched
Benchmarks are now launched using :

```
python -m enos.enos bench --workload=workload
```

enos will look into the workload directory for a file named
run.yml. This file is the description of the workload to launch.
One example is given below :

```
rally:
  enabled: true
  args:
    concurrency:
      - 1
      - 2
      - 4
    times:
      - 1
  scenarios:
    - name: boot and list servers
      enabled: true
      file: nova-boot-list-cc.yml
      args:
        sla_max_avg_duration: 30
        times : 5
```

This will launch all the scenarios described under the scenarios keys with all
the possible parameters. The parameters are calculated using the cartesian
product of the parameters given under the args keys. Locally defined args
(scenario level) shadow globally defined args (top level). The scenario must be
parameterized accordingly. The key (rally here) defines the type of benchmark to
launch : in the future we may support shaker, PerfKitBenchmarker ...

After all the possible combinations are run the environnment is backuped as
before. This is now done through a dedicated ansible role (backup).

fix #4

### CODE REVIEW

File: enos/enos.py [hunk 1]
```diff
@@ -306,33 +309,70 @@ def init_os(env=None, **kwargs):
...
+                for a in cartesian(top_args):
+                    playbook_path = os.path.join(ANSIBLE_DIR, 'run-bench.yml')
+                    inventory_path = os.path.join(SYMLINK_NAME, 'multinode')
+                    # NOTE(msimonin) all the scenarios must reside on the workload directory
+                    env['config']['scenario_type'] = nature
+                    env['config']['scenario'] = os.path.abspath(os.path.join(workload_dir, scenario["file"]))
+                    env['config']['scenario_args'] = a
+                    run_ansible([playbook_path], inventory_path, env['config'])
+    
+@enostask("""
```
Comment (rcherrueau) [inline enos/enos.py]: :+1:

File: workload/run.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+---
+# this is a sample of workload
+# prerequisite: 
+# - we rely on the rally template mechanism
+# 
+rally: 
+    enabled: true
```
Comment (rcherrueau) [inline workload/run.yml]: What is the purpose of the `enable` key at the top level? 

Maybe to prepare the field for other stress libraries, But, my philosophy is "If you don't use it, remove it".

File: enos/enos.py [hunk 3]
```diff
@@ -306,33 +309,70 @@ def init_os(env=None, **kwargs):
...
+        for nature, desc in workload.items():
+            scenarios = desc.get("scenarios", [])
+            for scenario in scenarios:
+                # merging args 
+                top_args = desc.get("args", {})
+                args = scenario.get("args", {})
+                top_args.update(args)
+                # merging enabled, skipping if not enabled
+                top_enabled = desc.get("enabled", True)
+                enabled = scenario.get("enabled", True)
```
Comment (rcherrueau) [inline enos/enos.py]: Do we wanna implicitly enable scenario, in this case giving the `True` default value is OK, or do we wanna explicitly enable scenario, in this case we should put `False` as the default value?

Comment (msimonin) [inline enos/enos.py]: good point, I would say : if you wrote the scenario, it's enabled.

Comment (rcherrueau) [inline enos/enos.py]: Great, keep it like that so

File: workload/run.yml (hunk 2 above)
Comment (msimonin) [inline workload/run.yml]: The purpose is to give the default value for all scenarios defined under the `scenarios` key.
In this example and according to the previous review we could omit it.

**Comment by msimonin** [comment]
Oh yes the readme :)

File: README.md [hunk 4]
```diff
@@ -103,19 +103,49 @@ python -m enos.enos -h
 > Using a virtualenv is encouraged
 
 
-## Launch rally benchmarks
+## Launch a workload
```
Comment (rcherrueau) [inline README.md]: :+1:

**Comment by msimonin** [comment]
I'll rebase

**Comment by msimonin** [comment]
Done

**Pull request merged**